crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -401,6 +409,10 @@ impl ConfigManagerActor {
                 error!("{}", error_message);
                 self.publish_command_status(ConfigOperation::Update(topic, request))
                     .await?;
+
+                // TODO: source temporary file should be cleaned up automatically
+                let _ = std::fs::remove_file(from);
```

Comment:
This has definitely to be done.

But not necessarily here. This file has been created and populated outside this method. It should be automatically cleaned by the caller. Not easy to implement with the current design, though. Can this be done once merged [this refactoring](https://github.com/thin-edge/thin-edge.io/pull/3086)